### PR TITLE
Update - Externally downloaded repos (requirements in armory repo but not in docker image)

### LINKED ISSUE #1302: untracked pygoturn dependency in carla baseline models
In [armory/baseline_models/pytorch/carla_goturn.py](https://github.com/twosixlabs/armory/blob/2b8dfe5925cb2563ce899e68f2555c129b00d410/armory/baseline_models/pytorch/carla_goturn.py#L6-L9) there is an import from a module which is required to be present but isn't in our source tree.

The problem is that pygoturn was never released on github nor on pypi so neither

    pip install pygoturn
    pip install git+https://github.com/amoudgl/pygoturn.git@1785ca7106fc0aa9e0fd03b440f34454e8a10e78 

work. 

The source repository hasn't had a commit since 2019, so I consider it effectively closed. I think the best approach is to fork the repo, add a pyproject.toml thus making the forked repo installable.

### PR DESCRIPTION
**Pull request by davidslater**
Fixes #1302 

Same as PR #1308 but merges in `develop`.

### CODE REVIEW

File: armory/errors.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import contextlib
```
Comment (mwartell) [inline armory/errors.py:1]: I really like the implementation of this as a context manager; nifty

I think errors.py is a confusing name for a module that imports, please change.

File: docs/style.md [hunk 2]
```diff
@@ -75,3 +75,22 @@ from armory.logs import log
...
+with ExternalRepoImport(
+    repo="colour-science/colour@v0.3.16",
+    experiment="carla_obj_det_dpatch_undefended.json",
+):
+    import colour
+```
+
+`repo` refers to the GitHub repo name (optionally with `@tag`).
+`experiment` refers to the `.json` file in the `scenario_config` directory that uses this repo.
+These repos are specifically *NOT* installed in the armory-supported docker containers and conda environments, and are downloaded at runtime.
```
Comment (mwartell) [inline docs/style.md:96]: A sentence of "why these are not downloaded" might be helpful.

**Review by mwartell — CHANGES_REQUESTED** [review]
Overall LGTM.

I think errors.py needs a rename so I've requested that change.

File: docs/style.md (hunk 2 above)
Comment (davidslater) [reply docs/style.md:96]: Added

**Comment by davidslater** [comment]
@mwartell ready for re-review. I moved the context manager to a different (existing) file: `armory.utils.external_repos`

**Pull request merged**